text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,431 @@
+# OpenTelemetry Telemetry Pipeline metrics
+
+Propose a uniform standard for telemetry pipeline metrics generated by
+OpenTelemetry SDKs and Collectors.
+
+## Motivation
+
+OpenTelemetry desires to standardize conventions for the metrics
+emitted by SDKs about success and failure of telemetry reporting. At
+the same time, the OpenTelemetry Collector has existing conventions
+which are expected to connect with metrics emitted by SDKs and have
+similar definitions.
+
+## Explanation
+
+This document proposes two metric instrument semantics threefour
+overall metric instruments for pipeline accounting, considering SDK
+and Collector pipeline metrics as separate, non-intersecting signals
+for reasons discussed below.  The three instruments are:
+
+- `otelcol_produced_items`: Exported, dropped, and discarded items (Collector)
+- `otelcol_consumed_items`: Received and inserted data items (Collector)
+- `otelsdk_produced_items`: Exported, dropped, and discarded items (SDK)
+
+The fourth instrument, named `otelsdk_consumed_items`, is logically
+well-defined by this proposal, but not necessary or useful, for
+SDK-specific reasons discussed below.
+
+We use the term "pipeline" to describe a directional arrangement of
+system components which produce, consume, and process telemetry on its
+way from the point of origin to the endpoint(s) in its journey.
+Pipeline components included in this specification are:
+
+- OpenTelemetry SDKs: As telemetry producers, these components are the
+  start of a pipeline.  These components also 
+- OpenTelemetry Collectors: The OpenTelemetry collector contains an
+  arrangement of components which act as both consumers and producers.
+  
+The terms "following"/"follower" and "preceding"/"preceder" refer to
+the relative components after or before another component in a
+pipeline.  The preceding component ("preceder") produces data that is
+consumed by the following component ("follower").
+
+An arrangement of pipeline components acting as a single unit, such as
```

Comment:
Is the intention that there will be similar `otelcol_*_items` metrics for the segments as well as the components?  It's not clear to me here how these two concepts apply here.

When it comes to "data loss", I am often more interested in the network boundary between "segments", e.g., when using the [loadbalancingexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/loadbalancingexporter) to route to a following Collector instance.
Currently, I compare the component level `loadbalancingexporter` and following `otlpreceiver` metrics to try to understand data loss, but really what I care about is sagment level view

Reply:
@0x006EA1E5, I'm working on writing out more details on data loss between segments.  [Here](https://github.com/open-telemetry/oteps/pull/259/files#r1619540718) is my current scribble that looks at how a resource exhausted response would look.